src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -26,45 +25,107 @@ public SignatureTrustAndValidityVerificationProvider()
         public Task<PackageVerificationResult> GetTrustResultAsync(ISignedPackageReader package, PrimarySignature signature, SignedPackageVerifierSettings settings, CancellationToken token)
         {
             token.ThrowIfCancellationRequested();
-
-            var result = VerifyValidityAndTrust(signature, settings);
+            var result = VerifySignatureAndCountersignature(signature, settings);
             return Task.FromResult(result);
         }
 
 #if IS_DESKTOP
-        private PackageVerificationResult VerifyValidityAndTrust(PrimarySignature signature, SignedPackageVerifierSettings settings)
+        private PackageVerificationResult VerifySignatureAndCountersignature(
+            PrimarySignature signature,
+            SignedPackageVerifierSettings settings)
+        {
+            var issues = new List<SignatureLog>();
+            var certificateExtraStore = signature.SignedCms.Certificates;
+            var primarySignatureHasCountersignature = SignatureUtility.HasRepositoryCountersignature(signature);
+
+            // Only accept untrusted root if the signature has a countersignature that we can validate against
```

Comment:
What is the reasoning behind this?

Reply:
In later waves this assertion will change by defaulting to false, unless the user specified the given signature source (either author or repository) as allowing it to be untrusted root. The reasoning behind allow this at this stage is to let an author signature have an untrusted root only if there is a countersignature on it that chains to a trusted root.